.github/workflows/promote.yaml

```diff
@@ -0,0 +1,36 @@
+name: Promote Charm
+
+on:
+  workflow_dispatch:
+    inputs:
+      promotion:
+        type: choice
+        description: Channel to promote from
+        options:
+          - edge -> stable
```

Comment:
Why directly from edge to stable? We should do edge -> candidate and candidate -> stable.